wiki/debugging.md

```diff
@@ -0,0 +1,76 @@
+
+### Step 3. Find the source URLs.
+
+The OCD API provides links to the Legistar API and web interface (look for the `source_url`). Open these, and check that the data in Legistar appears as expected.
+
+It does! Great, move onto the next step.
+
+It does not. Even better – contact Metro staff and let them know that the data discrepancy occurs in Legistar. Stop here.
+
+### Step 4. Go back to the OCD API. 
+
+Does the data appear as expected in the OCD API  (e.g., https://ocd.datamade.us/ocd-event/d29f59be-6f5b-41ad-958f-db0f63e245f2/)?
+
+It does! Great move onto the next step.
+
+It does not. The scrapers likely have a bug. Tracking down the exact nature of the bug may require some effort: fortunately, the scraper logs provide a good starting point. Shell into the server, and tail the metro log: 
+
+```bash
+ssh ubuntu@ocd.datamade.us
+
+tail -100 /tmp/lametro.log
+```
+
+Hopefully, the log offers some clues. An OperationalError indicates that something went awry with the database (e.g., not enough diskspace), and a ScraperError indicates a particular issue with the one of the scraper repos.
+
+If the scraper seems to be running without error, then the bug may be harder to find. A good first question: what are the most recent changes in the scraper codebase? could this have caused unusual behavior?
+
+### Step 5. Look at the Councilmatic database.
+
+Shell inside the `metro-councilmatic` server (`ssh ubuntu@lametro.datamade.us`). Query the database, and look for data inaccuracies. You can do this via postgres (database: `lametro`) or the Django shell: it's depends on the nature of the data problem, and your personal preferences. Is the data in the database?
+
+It is! Great, move on to the next step.
+
+It is not. The `import_data` command did not behave as expected. Tail the import logs, and look for clues. A common import error is the IntegrityError, which may alert you to missing or duplicate data: read the error carefully, and investigate the `ocd_id` that raised the error. 
+
+```bash
+tail -100 /var/log/councilmatic/lametro-importdata.log
+```  
+
+As with the scraper, if the import seems to be running without error, then the bug may be harder to find. A good first question: what are the most recent changes in the `django-councilmatic` codebase (the import script, but also the data modeling)? could this have caused unusual behavior?
+
+### Step 6. Investigate the display logic.
+
+You successfully confirmed that the data landed, without failure, in the Metro Councilmatic database! The bug resides somewhere in `la-metro-councilmatic` or the parent app, `django-councilmatic`. You can work through this by, first, examining the output of the view code, and then, inspecting how the templates deal with the data. Again, be sure to ask: what are the most recent changes, and could they have affected the health of the code?
+
+## Recurring problems
```

Comment:
it'd be good to list these known issues up high, even if it's just some nav items with links to the explanations down here.